cmd/neofs-adm/internal/modules/morph/container.go

```diff
@@ -104,6 +104,52 @@ func dumpContainers(cmd *cobra.Command, _ []string) error {
 	return os.WriteFile(filename, out, 0o660)
 }
 
+func listContainers(cmd *cobra.Command, _ []string) error {
+	c, err := getN3Client(viper.GetViper())
+	if err != nil {
+		return fmt.Errorf("can't create N3 client: %w", err)
+	}
+
+	inv := invoker.New(c, nil)
+
+	nnsCs, err := c.GetContractStateByID(1)
+	if err != nil {
+		return fmt.Errorf("can't get NNS contract state: %w", err)
+	}
+
+	var ch util.Uint160
+	s, _ := cmd.Flags().GetString(containerContractFlag)
+	if s == "" {
+		domain := containerContract + ".neofs"
+		ch, err = nnsResolveHash(inv, nnsCs.Hash, domain)
+		if err != nil {
+			return fmt.Errorf("unable to resolve hash for %s: %w", domain, err)
+		}
+	} else {
+		ch, err = util.Uint160DecodeStringLE(s)
+		if err != nil {
+			return fmt.Errorf("unable to decode container contract hash from arg %s: %w", containerContractFlag, err)
+		}
+	}
+
+	cids, err := unwrap.ArrayOfBytes(inv.Call(ch, "list", ""))
+	if err != nil {
+		return fmt.Errorf("%w: %v", errInvalidContainerResponse, err)
+	}
+
+	if len(cids) > 0 {
```

Comment:
We have only a loop inside which works fine. Do we need this condition?

Reply:
You right, redundant, forget to remove. Updated.